code/_core/obj/item/soulgems/_soulgem.dm

```diff
@@ -109,10 +129,13 @@
 		if(total_charge)
 			caller.visible_message(span("notice","\The [caller.name] recharges \the [S.name] with \the [src.name]."),span("notice","You charge \the [S] with \the [src]."))
 			S.total_charge += total_charge
+			total_charge -= total_charge
 			if(is_living(caller))
 				var/mob/living/L = caller
 				L.add_skill_xp(SKILL_MAGIC_ENCHANTING,CEILING(total_charge*0.025,1))
-			total_charge = 0
+			if(!is_star && total_charge == 0)
+				caller.to_chat(span("warning","\The [src] shatters!"))
+				qdel(src)
```

Comment:
total_charge -= total_charge should be removed, and the total_charge = 0 should be re-added.

if(!is_star && total_charge == 0) should just be (!is_star)